### DISCUSSION TITLE
How to delete Local Storage a/o IndexedDB for ALL Sites (not matching any rule)

### DISCUSSION THREAD

**Discussion by LaurentGrenet** [Cookie-AutoDelete/Cookie-AutoDelete#1415] [Q&A / Support]
Hi,

for any website for which a rule exists, it's quite easy to parameter this rule to keep only some cookie, but delete automatically all other cookies and choose to delete or to keep Local memory and/or IndexedDB and/or Cache and/or...

On the other hand, for website for which there is NO matching rule (neither White nor Grey), all cookies relative to this website are automatically cleaned by CAD when leaving the website.
But I've not been able to have Local Memory or IndexedDB or Cache cleaned for these websites (without any matching rule).

But maybe I didn't catch something...
So my question : Is there a mean to ask CAD to automatically delete Cache and/or Local data and/or IndexedDB and/or ... for all websites that do not match with any rule, on the same way it deletes all cookies for these sites. 

Thanks in advance for your help !
Regards
Laurent

**Comment by LaurentGrenet** [comment]
Sorry, I did a mistake.
For websites NOT matching any rule, local data and indexedDB seem to be correctly purged as well. Not only cookies.

I've been led to my mistake because it remain nevertheless (at least in some cases) A LOT of storage data for the site. But it seems to be another kind of data.
BTW if you could have a look here : https://support.mozilla.org/fr/questions/1381100 pls do not hesitate to add complementary explanations... if you have !

**Comment by kennethtran93** [comment]
Currently, the only way we can clear data from domains without it being opened is by keeping a cookie of the visited domain.  As long as there's a cookie for that domain, the other browsing site data for that domain will so be sent a cleanup request as well,unless configured otherwise. 

Thus if there are third-party domains that uses site data without cookies, we currently do not clean those as there's no cookies set for them yet. A request have been added to look into this, and may require additional permissions, notably the one that allows us to look into all web requests being made.
